pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -82,7 +84,6 @@ func (p *InstanceTypeProvider) getInstanceTypeZones(ctx context.Context) (result
 	if cached, ok := p.cache.Get(instanceTypeZonesCacheKey); ok {
 		return cached.(map[string]sets.String), nil
 	}
-	defer func() { p.cache.SetDefault(instanceTypeZonesCacheKey, result) }()
```

Comment:
I think this is bugged because I created a new map on line 87. If you use the named return type it should work. Otherwise remove the name from the named return type.